packages/replay-next/components/sources/hooks/useSearch.ts

```diff
@@ -271,10 +271,6 @@ export default function useSearch<Item, Result, QueryData = never>(
       dispatch({ type: "updateItems", items });
     }
 
-    if (queryChanged || queryDataChanged) {
-      dispatch({ type: "updateQuery", query: deferredQuery, queryData: deferredQueryData });
-    }
```

Comment:
This was the core of the bug. We shouldn't erase "global" query state when we encounter an invalid scope. We should just re-compute results for that scope (which we're already doing in this block).